src/Resources/config/twig_intl.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="sonata.templates.twig_intl.extension" class="Twig\Extra\Intl\IntlExtension">
```

Comment:
Do we need an own instance of this class? I prefer updating the docs and refer to the intl docs instead of registering a new instance.

Reply:
mmm what I can do is to only load this when the SonataIntlBundle is used to maintain BC and remove this in the next major version. And also update the docs to show how to use it.

Reply:
AFAIK, you should set the default of `use_intl_templates` to true to keep BC. So we don't need a BC layer here.

Reply:
I think the [intl templates are only loaded when SonataIntlBundle is enabled](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/DependencyInjection/AbstractSonataAdminExtension.php#L83). 

I'll be away some days, I'll continue when I come back or if someone wants to continue, feel free to do it.

Reply:
Here I am again! so I think these are the the scenarios here:
- The user doesn't have SonataIntlBundle installed:
    - Right now the bundle doesn't load anything.
    - `use_intl_templates: true` it will load the intl templates and register the service.
    - `use_intl_templates: false` it won't load anything.
- The user has SonataIntlBundle installed: 
    - Right now the bundle loads the intl templates and the SonataIntlBundle load the Twig extension that registers the twig functions needed.
    - `use_intl_templates: true` it will load the intl templates and register the service.
    - `use_intl_templates: false` it will also load the intl templates and register the service.

Reply:
So there is no change if you are still using `SonataIntlBundle`?

Reply:
It will load the intl templates but from the AdminBundle. We could throw a warning deprecation in case you have installed `SonataIntlBundle` but not enabled `use_intl_templates`.

I removed the definition and added the twig/extra-bundle so there is no need to register the service.